packages/adapter-vercel/files/entry.js

```diff
@@ -1,34 +1,43 @@
-import { getRawBody } from '@sveltejs/kit/node';
-
 // TODO hardcoding the relative location makes this brittle
 import { init, render } from '../output/server/app.js';
+import { assets } from './assets.js';
 
 init();
 
-export default async (req, res) => {
-	const { pathname, searchParams } = new URL(req.url || '', 'http://localhost');
-
-	let body;
-
-	try {
-		body = await getRawBody(req);
-	} catch (err) {
-		res.statusCode = err.status || 400;
-		return res.end(err.reason || 'Invalid request body');
+const getResult = (body, options) => ({
+	promise: Promise.resolve(),
+	waitUntil: Promise.resolve(),
+	response: new Response(body, options)
+});
+
+// eslint-disable-next-line
+_ENTRIES = {
+	middleware_kit: {
+		async default({ request }) {
+			if (request.method === 'GET') {
+				const { pathname, searchParams } = new URL(request.url);
+
+				if (!assets.has(pathname.slice(1))) {
```

Comment:
you could probably remove a level of nesting here, and a hard-to-grok negation:

```js
if (assets.has(pathname.slice(1))) {
  return
}
```

then the rest could be a level less-nested.

Reply:
sadly not — you need the `x-middleware-next` thing otherwise it will fail cryptically. so we'd either need to duplicate the `return` statement later on, or construct the URL before the main `if` block and add it to the condition there